src/main/java/com/google/devtools/build/lib/syntax/SkylarkSignatureProcessor.java

```diff
@@ -225,25 +225,31 @@ private static boolean isParamNamed(Param param) {
   }
 
   static Object getDefaultValue(Param param, Iterator<Object> iterator) {
+    return getDefaultValue(param.name(), param.defaultValue(), iterator);
+  }
+
+  static Object getDefaultValue(String paramName, String defaultValue, Iterator<Object> iterator) {
     if (iterator != null) {
       return iterator.next();
-    } else if (param.defaultValue().isEmpty()) {
-      return Runtime.NONE;
     } else {
-      try (Mutability mutability = Mutability.create("initialization")) {
-        // Note that this Skylark environment ignores command line flags.
-        Environment env =
-            Environment.builder(mutability)
-                .useDefaultSemantics()
-                .setGlobals(Environment.CONSTANTS_ONLY)
-                .setEventHandler(Environment.FAIL_FAST_HANDLER)
-                .build()
-                .update("unbound", Runtime.UNBOUND);
-        return BuildFileAST.eval(env, param.defaultValue());
-      } catch (Exception e) {
-        throw new RuntimeException(String.format(
-            "Exception while processing @SkylarkSignature.Param %s, default value %s",
-            param.name(), param.defaultValue()), e);
+      if (defaultValue.isEmpty()) {
```

Comment:
Why the change here?

Reply:
it's so that this method can be used by clients that have `ParamDescriptor` or `Param` annotation. It's somewhat unfortunate since having parameters of the same type is a bit confusing, but until all usages of annotations are removed this is the way to avoid duplication :(

Reply:
no problem at all, @c-parsons! Thanks for working on these changes! The only explanation I can think of with regards to your proxy overhead observation is that in the code you have used for benchmarking you didn't have a lot of parameters/methods. Since Buck does not have extension support, we end up having extremely complicated macros with hundreds (200+) parameters, so for us it ends up using a lot of cycles :( I'll gladly rebase my changes. Thanks again for working on Skylark perf and your reviews!

Reply:
btw, @c-parsons, what do you think about moving the parsed `defaultValue` to the `ParamDescriptor`? At the moment a lot of time is spent in a `LoadingCache`, so storing the variable directly next to a parameter can make for a potentially easier to use API and better for performance.